components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -481,10 +525,48 @@ func (r *NotebookReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return true
 		},
 	}
-	return c.Watch(
+
+	eventToRequest := handler.ToRequestsFunc(
+		func(a handler.MapObject) []ctrl.Request {
+			return []reconcile.Request{
+				{NamespacedName: types.NamespacedName{
+					Name:      a.Meta.GetName(),
+					Namespace: a.Meta.GetNamespace(),
+				}},
+			}
+		})
+
+	eventsPredicates := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			event := e.ObjectNew.(*v1.Event)
+			return e.ObjectOld != e.ObjectNew &&
+				isStsOrPodEvent(event) &&
+				nbNameExists(r.Client, nbNameFromInvolvedObject(&event.InvolvedObject), e.MetaNew.GetNamespace())
```

Comment:
Please @lluunn, take a look again. As you suggested and as we also discussed during the community meeting, https://github.com/kubeflow/kubeflow/pull/4139/commits/d41d01be3629e4ae89773bac38e66b8f582c74e2 checks if the event belongs to a notebook before triggering the reconcile call.

Reply:
thank you!